Feature: "Easy" view for media files

owehmer opened the pull request:
A first step towards #85. This allows a new user to toggle between a cleaner, grouped view and the standard view. In the future we could use this to implement an actual pro view and maybe (?) get rid of the standard view.

This changes the default starting page to the easy view. We might want to change that until we have a proper "first time landing page"

owehmer committed:
Added a basic media card

owehmer committed:
Group by media type pipe

owehmer committed:
Group names added

owehmer committed:
Added chips based on the attached tags / screens

owehmer committed:
Empty medias info added

owehmer committed:
removed unnecessary and reformat quotes

owehmer committed:
quotes part 2

negue commented on projects/contracts/src/lib/media.types.ts:
```diff
@@ -1,6 +1,7 @@
+import { Dictionary } from "./types";
 
 export enum MediaType {
+  Invalid = -1,
```
oh was it needed? lets hope nothing breaks haha

owehmer replied:
I mean I just wanted to have a common id for the invalid group and not some random 1337 number, that someone will eventualle use in that enum, cause thats when things start breaking.

negue replied:
maybe I should add more "defaults" to the enums so in that case its good

negue commented on src/app/manage/media/media-overview/group-by-media-type.pipe.spec.ts:
```diff
@@ -0,0 +1,60 @@
+import { GroupByMediaTypePipe, MediGroup } from './group-by-media-type.pipe';
+import { Clip, MediaType } from '@memebox/contracts';
+
+describe('GroupByMediaTypePipe', () => {
```
even tests !!! thank you

negue commented on src/app/manage/media/media-overview/media-overview.component.html:
```diff
@@ -30,20 +30,57 @@
+      <div>
+        <mat-icon inline svgIcon="info" color="primary" class="getting-started-info-pane__icon"></mat-icon>
+        <span>To add media files click on the
+          <button mat-fab color="warn" class="info-pane-inline-btn"><mat-icon svgIcon="add"></mat-icon></button>
```
could this be without pointer events ? otherwise I'm pretty sure some people would click and wonder why I didnt do anything :D

negue replied:
Also could you add like in `app-getting-started` an offline mode button (which will be used on github pages preview, since it doesn't have any server for it)

Once PR previews are added this will be useful

negue commented on src/app/manage/media/media-overview/media-card/media-card.component.html:
```diff
@@ -0,0 +1,86 @@
+    <button mat-button color="primary" (click)="onEdit.emit()">Edit</button>
+    <button mat-button (click)="onPreview.emit()">Preview</button>
+
+    <button mat-icon-button
```
could you add the `danger` color to it ?

negue reviewed (commented):
Oh awesome!! 🎉 

This will be a great addition to it.

The first time I opened I got a `this.media.tags is undefined` error. 

I dunno if its because of that error but the items no break into rows (for example I have multiple images).  
Guess no flex-wrap ?

Sorry for all the nagging here :D 

Optional no need to do:
- Could you also make group panels sticky? that its visible during scroll.
- Center the other types icons in the card ?

owehmer committed:
nullpointer fix

owehmer committed:
Disable ripple and hover on button

owehmer committed:
Styled buttons

owehmer committed:
Fill with dummy data

owehmer committed:
Flex wrap fixed

owehmer committed:
Sticky title

owehmer committed:
Replaced static colors with CSS variable

owehmer commented:
Don't worry :) I fixed everything and added the optional stuff aswell. Hope you like it!

Oh and I alseo replaced some of your static colors with a CSS Variable, that gets rid of a few TODOs :P

negue commented:
Awesome!!  the new UI looking gewd!! :)

Thank you for the huge feature / UI overhaul^^

negue commented:
@all-contributors add @owehmer for bug and code
